packages/video_player/video_player/lib/video_player.dart

```diff
@@ -558,9 +583,12 @@ class VideoPlayerController extends ValueNotifier<VideoPlayerValue> {
       return Caption.none;
     }
 
+    final delayedPosition = Duration(
+        microseconds:
+            position.inMicroseconds + value.captionOffset.inMicroseconds);
```

Comment:
It would be safer to add the `Duration`s and then convert to a specific unit afterward.